hdkeychain/extendedkey_test.go

```diff
@@ -893,3 +893,154 @@ func TestZero(t *testing.T) {
+			wantPriv:    "dprv3hCznBesA6jBuWH2xXKZSHtEkucFQFTaGhrXyuCCnNU24KDHHRsNcvAnq4596P5sYKTSgqbjUe2UNVxid8J9oDyFsuSWb3L93Jrka3vPtYV",
+			wantPrivSer: "12c0d59c7aa3a10973dbd3f478b65f2516627e3fe61e00c345be9a477ad2e215",
+			net:         mainNetParams,
+		},
+		{ // This test fails when using Child instead of ChildBIP32Std.
+			name:         "test vector 4 chain m/0H -- leading zeros in private key serialization",
+			master:       testVec4MasterHex,
+			path:         []uint32{hkStart},
+			wantPub:      "dpubZBYwK8RvCX3KjyCPhRQ9BYdX1ZYTvcFf67sRtoZGPnEYMCVgVyF5uWToDwePapvd1GtLYHAWGpQWo9ffod84MPvwpHmvKeZnZZf47EJcUmT",
+			wantPriv:     "dprv3jkqwzsd88yXqZJdSuLnfKqGPYzCVffataF79tcuPhdudeU3d1uUpvwn5Btmbai3BoVYhzbsEQ7tcH1XtEJ1BhzNSAtZfqv8BtSpdq5FEpj",
+			wantPrivSer:  "00d948e9261e41362a688b916f297121ba6bfb2274a3575ac0e456551dfd7f7e",
+			leadingZeros: 1, // 1 zero byte
+			net:          mainNetParams,
+		},
+		{
+			name:        "test vector 4 chain m/0H/1H -- completely different key",
+			master:      testVec4MasterHex,
+			path:        []uint32{hkStart, hkStart + 1},
+			wantPub:     "dpubZDVQT3iY5Bz6NtyCE8zfnpgwkN7AEUv5mr6H6vPVPRMupBbjTJojs7qRrvddW9adSJt6obdBgW6DsUAaFD8xRVQfVB4XfWCEYdTYRZ7updr",
+			wantPriv:    "dprv3mhK5vAEzovJUV5RycwKGbth8MYtoYL1aJTxN1T8PLmH6da6aMU8nYKQiC5Zj7mC7pH4G8xPxGZVR51FG9Ssptzx1c39A1JxADS6zDrsLPr",
+			wantPrivSer: "3a2086edd7d9df86c3487a5905a1712a9aa664bce8cc268141e07549eaa8661d",
+			net:         mainNetParams,
+		},
+	}
+
+tests:
+	for i, test := range tests {
+		masterSeed, err := hex.DecodeString(test.master)
+		if err != nil {
+			t.Errorf("DecodeString #%d (%s): unexpected error: %v",
+				i, test.name, err)
+			continue
+		}
+
+		extKey, err := NewMaster(masterSeed, test.net)
+		if err != nil {
+			t.Errorf("NewMaster #%d (%s): unexpected error when "+
+				"creating new master key: %v", i, test.name,
+				err)
+			continue
+		}
+
+		for _, childNum := range test.path {
+			var err error
+			extKey, err = extKey.ChildBIP32Std(childNum)
+			if err != nil {
+				t.Errorf("err: %v", err)
+				continue tests
+			}
+		}
+
+		priv, err := extKey.SerializedPrivKey()
+		if err != nil {
+			t.Errorf("SerializedPrivKey #%d (%s): unexpected error: %v",
+				i, test.name, err)
+			continue
+		}
+		if len(priv) != 32 {
+			t.Errorf("serialized private key length %d, want 32", len(priv))
+		}
+
+		privStr := hex.EncodeToString(priv)
+		if privStr != test.wantPrivSer {
+			t.Errorf("Serialize #%d (%s): mismatched serialized "+
+				"private extended key -- got: %s, want: %s", i,
+				test.name, privStr, test.wantPrivSer)
+			continue
+		}
+
+		extKeyStr := extKey.String()
+		if extKeyStr != test.wantPriv {
+			t.Errorf("Serialize #%d (%s): mismatched serialized "+
+				"private extended key -- got: %s, want: %s", i,
+				test.name, extKeyStr, test.wantPriv)
+			continue
+		}
+
+		pubKey := extKey.Neuter()
+
+		// Neutering a second time should have no effect.
+		// Test for referencial equality.
```

Comment:
```suggestion
		// Test for referential equality.
```

Reply:
Fixed here an in `TestBIP0032Vectors` at the top of the file.